### ISSUE TITLE
Console players unable to connect

### ISSUE THREAD

**Issue by mastachef** [GeyserMC/Geyser#4243]
### Describe the bug

Players on switch, playstation and xbox are reporting this error that is not allowing them to join the server

### To Reproduce

Use console and connect on 1.20.40

### Expected behaviour

Join normally

### Screenshots / Videos

![20231024_165136](https://github.com/GeyserMC/Geyser/assets/80359894/7992051f-119a-478c-a79b-4c9d445bce32)
![Minecraft_2023-10-24-15-45-36](https://github.com/GeyserMC/Geyser/assets/80359894/b6a833d7-02ba-432b-9008-4ea13a558f6e)

https://github.com/GeyserMC/Geyser/assets/80359894/cadf2104-ca3b-4249-bb8e-7206e72268bc

### Server Version and Plugins

velocity-3.2.0-SNAPSHOT-276 

git-Pufferfish-17 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: f6a081a on HEAD)

### Geyser Dump

https://dump.geysermc.org/lmSlQBzTd5lWpT9h7e4D9lLh3QhFhbwj

### Geyser Version

Geyser version 2.2.0-SNAPSHOT (git-master-2e68244) (Java: 1.20.2, Bedrock: 1.20.0/1.20.1 - 1.20.40)

### Minecraft: Bedrock Edition Device/Version

1.20.40

**Comment by ch4ika** [comment]
I can confirm.

damaged package
![56efb102-038a-4ae1-aaba-ca9fd17eea73](https://github.com/GeyserMC/Geyser/assets/104507238/2466c556-a3a2-45b3-9cfc-46356a09603e)

**Comment by iMythJj** [comment]
My friend on Xbox Series S can join with the new Geyser snapshot, but my friend who plays on PS5 cannot join anymore. His game crashes every time he tries to join.

**Comment by Leahcimkrob** [comment]
It may be that your colleague has not yet had the update to 1.20.40 on the xBox,

**Comment by mastachef** [comment]
> It may be that your colleague has not yet had the update to 1.20.40 on the xBox,

All examples were from 1.20.40

**Comment by Leahcimkrob** [comment]
I mean iMythJj

**Comment by Shaylin-8bit** [comment]
Can say the same here. Spigot server running 1.20.1 with latest Geyser and latest ViaVersion. 
Both PS4 and Android 1.20.40 devices are receiving error messages when joining.

**Comment by Leahcimkrob** [comment]
Same with geyser on Bungee (Build 1759) and Viaversion (Build 378) ViaBackwards (Build 349) on Paper 1.20.1 Build 195
Logn with every Bedrock 1.20.40 device not possible
![Unbenannt](https://github.com/GeyserMC/Geyser/assets/70509359/8138a657-89f9-4452-aacc-9d02bbfb8834)

**Comment by julienc82** [comment]
Same here. It crashes on my son's account, but not mine. Note that yesterday after updating my server, I connected with 1.20.32, and that I had an invisibility potion effect without asking anything. I had to make a command to remove all the effects. I then updated my Minecraft client and on this account it works fine.

**Comment by jmbullet** [comment]
My friend on PS4 is having the same issue. When he told me about it, I updated the server and then I updated all the plugins but it's still not working. Java players can join without any issue. Weirdly enough, I can join the same server with my phone on the android version and my computer on the bedrock version (same microsoft account) just fine. My phone is on 1.20.40

**Comment by Konicai** [comment]
Possible duplicate of #4240

**Comment by onebeastchris** [comment]
There are multiple issues here, but likely all from the same core issue:
- "Server sent broken packet" (or the localized equivalent) is usually the result of whatever console connecting method being outdated (e.g.: using an outdated BedrockConnect dns, or an outdated BedrockTogether/similar app that broadcasts a local world for consoles to join). For this, the solution is fairly straightforward - use an updated method to connect to servers; for example by updating FriendConnect (or using the official BedrockConnect DNS, which is [updated to 1.20.40](https://github.com/Pugmatt/BedrockConnect/commit/92f1e0e261bf527d0e2127639ccaac0af5766001))

- "This world is no longer available to join" is usually the result of e.g. the BedrockTogether app being shut down while trying to connect to the server. In some cases, the local server/friend popup persists longer than it should.

- Crashing upon joining is a known issue as konicai linked, but should be unrelated to the errors here.

In short: When a new Bedrock version releases, console players must also update their console joining workaround. Ideally, a proper disconnect message would show that, but we can't implement that as the connection never gets to Geyser itself.
